File: selfstudy/golang/cars_sortdesc.go
/*
task 230:
Cars - SortDesc
You need to sort the values in descending order for car speeds in km/h
*/

package main

import (
	"fmt"
	"reflect"
)

func Sort_downCars(speeds []int) []int {
	res := make([]int, len(speeds))
	copy(res, speeds)
	for i := 0; i < len(res); i++ {
		for j := i+1; j < len(res); j++ {
			if res[i] < res[j] { res[i], res[j] = res[j], res[i] }
		}
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Sort_downCars([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Sort_downCars([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Sort_downCars([]int{1, 2, 3})},
		{[]int{100, 200}, Sort_downCars([]int{100, 200})},
		{[]int{}, Sort_downCars([]int{})},
	}
	
	for idx, t := range tests {
		res := Sort_downCars(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// works i think

File: selfstudy/python/cars_sortdesc.py
'''
task 130:
Cars - SortDesc
You need to sort the values in descending order for car speeds in km/h
'''

def sort_down_cars(speeds):
    return sorted(speeds, reverse=True)

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], sort_down_cars([10, 20, 30])),
        ([50, 60, 70], sort_down_cars([50, 60, 70])),
        ([1, 2, 3], sort_down_cars([1, 2, 3])),
        ([100, 200], sort_down_cars([100, 200])),
        ([], sort_down_cars([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = sort_down_cars(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# works i think
